AdyenCard/Components/Card/CardViewController.swift

```diff
@@ -110,36 +111,18 @@ internal class CardViewController: FormViewController {
     internal var cardBIN: String {
         items.numberContainerItem.numberItem.binValue
     }
-    
+
     internal var validAddress: PostalAddress? {
-        guard let address = address, isAddressValid(address: address) else { return nil }
-        return address
-    }
-    
-    private func isAddressValid(address: PostalAddress) -> Bool {
-        let fieldsValues: [String?]
-        
-        switch configuration.billingAddress.mode {
-        case .full:
-            fieldsValues = [address.city,
-                            address.country,
-                            address.postalCode,
-                            address.stateOrProvince,
-                            address.street,
-                            address.houseNumberOrName]
-        case .postalCode:
-            fieldsValues = [address.postalCode]
-        case .none:
-            fieldsValues = []
-        }
-        
-        let trimmedFieldsValues = fieldsValues.map {
-            $0?.trimmingCharacters(in: .whitespaces).adyen.nilIfEmpty
+        guard let address = address,
+              AddressValidator.init(addressMode: configuration.billingAddress.mode,
+                                    address: address,
+                                    addressScheme: items.addressViewModel?.scheme).isValid() else {
```

Comment:
have you considered making the `address ` parameter passed to the `isValid()` function instead of the initializer? this way we can reuse it? 

also I am wondering why do use the `AddressValidator.init` instead of just `AddressValidator()`